### ISSUE TITLE
Restart is not working - Still having ECONNREFUSED

### ISSUE THREAD

**Issue by mfornasa** [prerender/prerender#498]
I am experiencing the same `Error: connect ECONNREFUSED 127.0.0.1:9222` described in this closed issue: https://github.com/prerender/prerender/issues/446.  I am on version 5.2.0, which contains the restart patch. The system is subject to high traffic.

Chrome restart happens correctly on a staging system configured in the same way, but with less traffic.

I see that there is a new patch for restart https://github.com/prerender/prerender/commit/b0426f17dd8fa61d5b3ec8afcd46b2b81b132c63, but I am not sure if this could address my issue.

**Comment by thoop** [comment]
If you have a single large server with lots of traffic, you might run into these issues. We've had better performance with multiple smaller servers behind a load balancer to keep the traffic to each Chrome instance at a manageable rate.

We do want to improve this though if the traffic isn't too much for Chrome to handle. On average, how may requests per minute are you sending to your server and how long does each page take to load?

**Comment by mfornasa** [comment]
@thoop I already have multiple prerender instances. Traffic could be around 10-20 rpm for each rpm probably. Is that too much?
But if I understand correctly, chrome should be forced to restart each hour. This is not happening. Is that expected? Is https://github.com/prerender/prerender/commit/b0426f17dd8fa61d5b3ec8afcd46b2b81b132c63 related to this in any way?

**Comment by thoop** [comment]
No, https://github.com/prerender/prerender/commit/b0426f17dd8fa61d5b3ec8afcd46b2b81b132c63 wouldn't be related. That was just a small error case where a request might try to be loaded during the restart process.

The Prerender server will try to restart Chrome after 10 minutes if there are zero requests in flight. If the Prerender server is unable to restart Chrome after 60 minutes of that (so a request was always in flight), it will delay the incoming requests while it restarts Chrome.

Are you not seeing "Restarting Chrome" anywhere in your Prerender server logs?

**Comment by mfornasa** [comment]
The “Restarting Chrome” appears in the logs, but I can see from “ps” that
Chrome is not being restarted.

**Comment by thoop** [comment]
Which operating system are you using? Are you just seeing the pid stay the same after a restart? Or how can you tell that it's not being restarted?

When you see "Restarting Chrome", do you see all of the restart process like:

```
2018-03-27T13:38:52.163Z Restarting Chrome
2018-03-27T13:38:52.191Z Chrome connection closed... restarting Chrome
2018-03-27T13:38:52.192Z Starting Chrome
2018-03-27T13:38:52.702Z Started Chrome: HeadlessChrome/65.0.3325.181
```

Thanks for providing any info!

**Comment by mfornasa** [comment]
Logs are:
```
2018-03-27T14:18:37.153Z Restarting Chrome
2018-03-27T14:18:37.164Z Chrome connection closed... restarting Chrome
2018-03-27T14:18:37.164Z Starting Chrome
2018-03-27T14:18:37.669Z Started Chrome: HeadlessChrome/65.0.3325.181
```
However, Chrome is not restarted, verified by using `ps aux` `START` field.

OS is Centos 7 in a Docker container.

This does not happen all the time. I suspect that it happens when there is higher traffic.

**Comment by thoop** [comment]
Hi @mfornasa. Do you have any updates here so we can look into this issue more?

**Comment by Neel9722** [comment]
Hi @thoop i am getting same error of "timed out waiting for chrome connection".

Server restart will make it work but i happens again and again after some interval.

please help.

**Comment by thoop** [comment]
@Neel9722 I responded to your other issue.
